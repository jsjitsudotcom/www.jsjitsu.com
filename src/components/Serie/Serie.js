import React from "react";
import Style from "./Serie.scss";
import Ripple from "./../Ripple/Ripple";

const Serie = ({ illustration, title }) => (
  <div className={Style.container}>
    <Ripple color="rgba(255,255,255, 0.5)" during={1000}>
      <div
        className={Style.illustration}
        style={{ backgroundImage: `url(${illustration})` }}
      />
      <div className={Style.title}>{title}</div>
    </Ripple>
  </div>
);

export default Serie;
