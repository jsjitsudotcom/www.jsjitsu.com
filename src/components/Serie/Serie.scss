:local(.container) {
  width: 100px;
  background-color: #212121;
  border-radius: 4px;
  box-shadow: 0 2px 2px rgba(black, 0.5);
  overflow: hidden;
  cursor: pointer;
}

:local(.illustration) {
  width: 100%;
  height: 145px;
  background-size: cover;
  background-position: 50% 50%;
}

:local(.title) {
  padding: 10px;
  color: white;
}
