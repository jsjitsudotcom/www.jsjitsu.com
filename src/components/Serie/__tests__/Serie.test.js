import React from "react";
import { mount, shallow } from "enzyme";
import { Provider } from "react-redux";
import fetchMock from "fetch-mock";
import sinon from "sinon";

import Serie from "./../Serie";

describe("<Serie />", () => {
  it("Le composant doit être rendu avec mount", () => {
    shallow(<Serie />);
  });

  it("Doit vérifier la bon affichage des props", async () => {
    const wrapper = shallow(<Serie illustration="illustration" name="name" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Doit appeller la méthode onClick", async () => {
    const spy = sinon.spy();
    const wrapper = shallow(<Serie onClick={spy} />);

    wrapper.find("Ripple").prop("onClick")();

    const data = spy.called;
    const expected = true;
    expect(data).toEqual(expected);
  });
});
