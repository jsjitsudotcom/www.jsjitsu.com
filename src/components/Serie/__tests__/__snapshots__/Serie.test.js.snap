// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Serie /> Doit v√©rifier la bon affichage des props 1`] = `
<div
  className="container"
>
  <Ripple
    color="rgba(255,255,255, 0.5)"
    during={1000}
    onClick={[Function]}
  >
    <div
      className="illustration"
      style={
        Object {
          "backgroundImage": "url(illustration)",
        }
      }
    />
    <div
      className="title"
    >
      name
    </div>
  </Ripple>
</div>
`;
